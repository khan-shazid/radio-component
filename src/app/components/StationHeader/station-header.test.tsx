import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Header from './station-header';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header/>, div);
})

it('renders header correctly', () => {
    const {getByTestId} = render(<Header/>);
    expect(getByTestId('header')).toHaveTextContent('STATIONS');
})

it('matches snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
})
