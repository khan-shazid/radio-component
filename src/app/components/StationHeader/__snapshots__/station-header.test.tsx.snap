// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="stationHeader"
  data-testid="header"
>
  <div
    className="back"
  />
  <div
    className="title"
  >
    STATIONS
  </div>
  <div
    className="signOut"
  />
</header>
`;
